demos/common/performance_metrics/metrics.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "metrics.h"
+
+#include <iostream>
+#include <limits>
+#include <sstream>
+#include <iomanip>
+
+PerformanceMetrics::PerformanceMetrics(Duration timeWindow) {
+    // 'timeWindow' defines the length of the timespan over which the 'current fps' value is calculated
+    timeWindowSize = timeWindow;
+
+    lastMovingStatistic = Statistic();
+    currentMovingStatistic = Statistic();
+    totalStatistic = Statistic();
+}
+
+void PerformanceMetrics::update(TimePoint lastRequestStartTime,
+                                cv::Mat& frame,
+                                cv::Point position,
+                                double fontScale,
+                                cv::Scalar color,
+                                int thickness) {
+    TimePoint currentTime = Clock::now();
+
+    if (!lastUpdateTime.time_since_epoch().count()) {
+        lastUpdateTime = currentTime;
+        return;
+    }
+    
+    currentMovingStatistic.latency += currentTime - lastRequestStartTime;
+    currentMovingStatistic.period = currentTime - lastUpdateTime;
+    currentMovingStatistic.frameCount++;
+
+    if (currentTime - lastUpdateTime > timeWindowSize) {
+        lastMovingStatistic = currentMovingStatistic;
+        totalStatistic.combine(lastMovingStatistic);
+        currentMovingStatistic = Statistic();
+
+        lastUpdateTime = currentTime;
+    }
+
+    // Draw performance stats over frame
+    Metrics metrics = getLast();
+    std::ostringstream out;
+    if (metrics.latency == metrics.latency) {
+        out.str("");
```

Comment:
Turning `out` into a field will make the class non-copy-constructible, which is not quite handy. Is it critical, in terms of run time, to avoid creating `ostringstream` each time here?

Reply:
May be we could specify `timeWindow` as template. That should eliminate a need to define constructor for `PerformanceMetrics` and compiler will generate move constructor for you. But I think it's simpler to create `ostringstream` each time.